modules/php/States/DrawCardsTrait.php

```diff
@@ -12,18 +12,19 @@ public function st_drawCards()
     $cards = $game->cards;
 
     // Check if this player is empty handed and the "no-hand-bonus" is in play
-    $hasNoHandBonus =
-      count($cards->getCardsOfTypeInLocation("rule", 216, "rules")) > 0;
+    $hasNoHandBonus = Utils::getActiveNoHandBonus();
     $cardsInHand = $cards->countCardInLocation("hand", $player_id);
 
+    $addInflation = Utils::getActiveInflation() ? 1 : 0;
+    $partyBonus = Utils::isPartyInPlay() ? 1 + $addInflation : 0;
```

Comment:
I think you need to add `getActivePartyBonus` here?

Reply:
yes, of course, absolutely right